lua/autorun/server/sv_spawn_protection.lua

```diff
@@ -259,3 +265,26 @@ hook.Add( "KeyPress", "CFCspawnProtectionKeyPressCheck", spawnProtectionKeyPress
 
 -- Prevent entity damage while in spawn protection
 hook.Add( "EntityTakeDamage", "CFCpreventDamageDuringSpawnProtection", preventDamageDuringSpawnProtection, HOOK_HIGH )
+
+-- Disable weapon checking on a player once they die
+hook.Add( "PlayerDeath", "CFCspawnProtectionPauseCheckingWeapons", function( ply )
+    ply.spawnProtectionCannotCheckWeapons = true
+end )
+
+-- Resume weapon checking on a player with a slight delay after they spawn, after loadout addons give their weapons
+-- The expectation is for loadouts to prioritize holding the physgun at the end, making players not lose spawn protection
+-- Note that some loadout addons (like StyledStrike's) use a 0.1s timer on PlayerLoadout to give weapons
+hook.Add( "PlayerSpawn", "CFCspawnProtectionResumeCheckingWeapons", function( ply )
```

Comment:
This arbitrary timer can be avoided, use `PlayerSetModel`  it's called after PlayerLoadout which any loadout addon will use, https://github.com/Facepunch/garrysmod/blob/39867454ca05ffaa04d70e74a5aac6f0858a9272/garrysmod/gamemodes/base/gamemode/player.lua#L250

Reply:
Nope, StyledStrike's addon uses a timer.Simple()
<https://github.com/StyledStrike/gmod-custom-loadout/blob/master/lua/custom_loadout/sv_main.lua#L93>